lib/upipe-freetype/upipe_freetype.c

```diff
@@ -170,33 +219,25 @@ static void upipe_freetype_input(struct upipe *upipe, struct uref *uref, struct
 {
     struct upipe_freetype *upipe_freetype = upipe_freetype_from_upipe(upipe);
 
+    if (!upipe_freetype->ubuf_mgr) {
+        upipe_freetype_demand_ubuf_mgr(upipe,
+                                       uref_dup(upipe_freetype->flow_output));
+        if (unlikely(!upipe_freetype->ubuf_mgr)) {
+            upipe_warn(upipe, "no ubuf manager provided, drop...");
```

Comment:
The correct way to handle this is to use helper_input and buffer until a ubuf manager is provided.